hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AbstractAmqpErrorHandler.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.amqp;
+
+import org.springframework.amqp.AmqpRejectAndDontRequeueException;
+
+/**
+ * An abstract error handler for errors resulting from AMQP.
+ */
+public abstract class AbstractAmqpErrorHandler<T> implements AmqpErrorHandler{
+
+    @Override
+    public void doHandle(Throwable throwable, AmqpErrorHandlerChain chain) {
+        final Throwable cause = throwable.getCause();
+        if (getExceptionClass().isAssignableFrom(cause.getClass())) {
+            throw new AmqpRejectAndDontRequeueException(getErrorMessage(throwable));
+        } else {
+            chain.handle(throwable);
+        }
+    }
+
+    /**
+     * Returns the class of the exception.
+     *
+     * @return
+     *          the exception class
+     */
+    public abstract Class<T> getExceptionClass();
+
+    /**
+     * Returns the customized error message.
+     *
+     * @return
+     *          the customized error message
+     */
+    public String getErrorMessage(Throwable throwable){
+      return new AmqpFormulateErrorMessage().constructErrorMessage(throwable);
```

Comment:
First of all it is unnecessary to create and instance of `AmqpFormulateErrorMessage` each time we call `getErrorMessage`, I guess `constructErrorMessage` can be static, isn't it?

Reply:
Additionally, I would rename `AmqpFormulateErrorMessage ` to something like `AmqpErrorMessageExtractor` or `AmqpErrorMessageFormater` as class names should not use verbs

Reply:
Agreed on both the points, can make the method static and also will name `AmqpFormulateErrorMessage `  class to `AmqpErrorMessageComposer ` as this class at the moment is mainly composing a meaningful message